grovedb/src/util.rs

```diff
@@ -149,6 +149,117 @@ macro_rules! storage_context_with_parent_optional_tx {
+                                    let $root_key = root_key;
+                                    let $is_sum_tree = true;
+                                    $($body)*
+                                    }
+                                    _ => {
+                                        return Err(Error::CorruptedData(
+                                            "parent is not a tree"
+                                                .to_owned(),
+                                        )).wrap_with_cost($cost);
+                                    }
+                            }
+                        },
+                        Err(e) => Err(e),
+                    }
+                } else {
+                    return Err(Error::CorruptedData(
+                        "path is empty".to_owned(),
+                    )).wrap_with_cost($cost);
+                }
+            } else {
+                let $storage = $db
+                    .get_storage_context($path.clone(), $batch).unwrap_add_cost(&mut $cost);
+                if let Some((parent_path, parent_key)) = $path.derive_parent() {
+                    let parent_storage = $db.get_storage_context(
+			            parent_path,
+                        $batch
+		            ).unwrap_add_cost(&mut $cost);
+                    let result = Element::get_from_storage(&parent_storage, parent_key)
+                        .map_err(|e| {
+                            Error::PathParentLayerNotFound(
+                                format!(
+				                    "could not get key for parent of subtree optional no tx: {}",
+				                    e
+				                )
+                            )
+                        }).unwrap_add_cost(&mut $cost);
+                    match result {
+                        Ok(element) => {
+                            match element {
+                                Element::Tree(root_key, _) => {
+                                    let $root_key = root_key;
+                                    let $is_sum_tree = false;
+                                    $($body)*
+                                }
+                                Element::SumTree(root_key, ..) => {
+                                    let $root_key = root_key;
+                                    let $is_sum_tree = true;
+                                    $($body)*
+                                    }
+                                    _ => {
+                                        return Err(Error::CorruptedData(
+                                            "parent is not a tree"
+                                                .to_owned(),
+                                        )).wrap_with_cost($cost);
+                                    }
+                            }
+                        },
+                        Err(e) => Err(e),
+                    }
+                } else {
+                    return Err(Error::CorruptedData(
```

Comment:
doesn't seem like corrupted data but a wrong arg to macro

Reply:
Agreed, since this is just copied code we should make a new PR to fix this in all places.